hack/devtools/deploy-shared-env.sh

```diff
@@ -88,7 +89,8 @@ deploy_oic_for_dedicated_rp() {
         -n rp-oic \
         --template-file pkg/deploy/assets/rp-oic.json \
         --parameters \
-            "rpServicePrincipalId=$(az identity show -g $RESOURCEGROUP -n aro-rp-$LOCATION | jq -r '.["principalId"]')"
+            "rpServicePrincipalId=$(az identity show -g $RESOURCEGROUP -n aro-rp-$LOCATION | jq -r '.["principalId"]')" \
+            "storageAccountDomain=$(yq '.rps[].configuration.storageAccountDomain' dev-config.yaml | cut -d '.' -f1)"
```

Comment:
Two questions here:

1. Could you run me through why `deploy_oic_dev` uses the resource group name but `deploy_aks_dev` uses `yq` to grab it from a config file? I think you intended to address this in the PR description, but I'm missing some connections between the dots.
2. Answering Q1 may help me understand this one as well, but isn't this shell stuff (`yq` output piped to `cut`) duplicating the changes made in `pkg/deploy/generator/resources_oic.go`? Why is that?

Reply:
For your first question, we have 3 different environments to account for:
1) Prod - gets the storage account name from the storageAccountDomain, which is in RP-Config
2) Full RP service in dev - gets the storage account name from the storageAccountDomain, which is in dev-config.yaml (basically the same behavior as 1, or as similar as we can get)
3) Shared dev / local dev - there isn't a dev-config/RP-config available so we have to get the name from somewhere else. Since we need unique names, the only clear path forward that I can see is to use the resource group name, but I'm definitely open to ideas.

For your second question, good catch. I'll change to just `yq '.rps[].configuration.storageAccountDomain' dev-config.yaml` which should mirror the prod behavior more.